### ISSUE TITLE
Feature Request: Change Accessory Type

### ISSUE THREAD

**Issue by nyxs** [nitaybz/homebridge-delay-switch#63]
Hey,
I think it would be nice if we could decide what would the type of the switch would be.
For example: I want my switch to be a light.

why?
Because then I would be able to group is with the actual light that is being controlled by the delay switch.

Scenario:
I have the kitchen light, for example.
When I detect motion in the kitchen, in certain conditions, I’m turning on the delay switch, which turns on the actual light.
When the delay switch turns off, the kitchen light turns off too.
Today, to have this sync between the switch and the light, I have to do 4 automations for each switch:
1. Turn on the light when the switch is on.
2. The opposite, to turn off.
3. Turn on the switch when the light turns on.
4. The opposite, to turn off.

basically, Homekit supports grouping, which means that if one accessory in the group goes to a specific status, the other accessories would change to that status.
Logically, they can do that only between the same types. We cannot group a switch and a light.
Hence, if we could change the type of the switch the other types, we would not need 4 automations for each switch-light combination.

Hope you’ll find it necessary and implement this small change.

thanks

**Comment by nitaybz** [comment]
Switch can be changed to Light pretty easily in Home app in the accessory settings

**Comment by nyxs** [comment]
I know that usually there’s an option for that, but I checked the switches from the plugin, and they don’t have this option

**Comment by ursusHU** [comment]
> I know that usually there’s an option for that, but I checked the switches from the plugin, and they don’t have this option

Yes they have. It can be light, fan, switch. You have to open the group (I assume you have a sensor too) and modify the icon of the switch.

**Comment by nyxs** [comment]
Ohh, ok, just saw it now.
Thanks!
